futures-util/src/stream/stream/unzip.rs

```diff
@@ -0,0 +1,66 @@
+use core::mem;
+use core::pin::Pin;
+use futures_core::future::{FusedFuture, Future};
+use futures_core::stream::{FusedStream, Stream};
+use futures_core::task::{Context, Poll};
+use pin_project::pin_project;
+
+/// Future for the [`unzip`](super::StreamExt::unzip) method.
+#[pin_project]
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Unzip<St, FromA, FromB> {
+    #[pin]
+    stream: St,
+    left: FromA,
+    right: FromB,
+}
+
+impl<St: Stream, FromA: Default, FromB: Default> Unzip<St, FromA, FromB> {
+    fn finish(self: Pin<&mut Self>) -> (FromA, FromB) {
+        let this = self.project();
+        (
+            mem::take(this.left),
+            mem::take(this.right),
```

Comment:
```suggestion
            mem::replace(this.left, Default::default()),
            mem::replace(this.right, Default::default()),
```
(`mem::take` requires Rust 1.40.)